probe_scraper/runner.py

```diff
@@ -118,49 +121,52 @@ def write_general_data(out_dir):
         )
 
 
-def write_glean_metric_data(metrics, dependencies, out_dir):
-    # Save all our files to "outdir/glean/<repo>/..." to mimic a REST API.
+def write_glean_metric_data(
+    metrics: Dict[str, Any], dependencies: Dict[str, Any], out_dir: Path
+):
+    # Save all our files to "out_dir/glean/<repo>/..." to mimic a REST API.
     for repo, metrics_data in metrics.items():
         dependencies_data = dependencies[repo]
 
-        base_dir = os.path.join(out_dir, "glean", repo)
+        base_dir = out_dir / "glean" / repo
 
         dump_json(general_data(), base_dir, "general")
         dump_json(metrics_data, base_dir, "metrics")
         dump_json(dependencies_data, base_dir, "dependencies")
 
 
-def write_glean_tag_data(tags, out_dir):
-    # Save all our files to "outdir/glean/<repo>/..." to mimic a REST API.
+def write_glean_tag_data(tags: Dict[str, Any], out_dir: Path):
+    # Save all our files to "out_dir/glean/<repo>/..." to mimic a REST API.
     for repo, tags_data in tags.items():
-        base_dir = os.path.join(out_dir, "glean", repo)
+        base_dir = out_dir / "glean" / repo
         dump_json(tags_data, base_dir, "tags")
 
 
-def write_glean_ping_data(pings, out_dir):
-    # Save all our files to "outdir/glean/<repo>/..." to mimic a REST API.
+def write_glean_ping_data(pings: Dict[str, Any], out_dir: Path):
+    # Save all our files to "out_dir/glean/<repo>/..." to mimic a REST API.
     for repo, pings_data in pings.items():
-        base_dir = os.path.join(out_dir, "glean", repo)
+        base_dir = out_dir / "glean" / repo
         dump_json(pings_data, base_dir, "pings")
 
 
-def write_repositories_data(repos, out_dir):
+def write_repositories_data(repos: List[Repository], out_dir: Path):
     json_data = [r.to_dict() for r in repos]
-    dump_json(json_data, os.path.join(out_dir, "glean"), "repositories")
+    dump_json(json_data, out_dir / "glean", "repositories")
```

Comment:
Have you checked the content of `repositories` file generated here? I'm not sure why but when this failed on Wednesday this file was written as binary.

Reply:
Yes, i checked the content and it matches what is produced on `main`. It is binary because it's gzip compressed, per https://github.com/mozilla/probe-scraper/pull/441/files#diff-b888a53e56d384dcd1d951e902d7957952d441c3d4abfc91eb164a69e5f8eb5fR471-R474 :
> ```
>         # cloudfront is supposed to automatically gzip objects, but it won't do that
>         # if the object size is > 10 megabytes (https://webmasters.stackexchange.com/a/111734)
>         # which our files sometimes are. to work around this, we'll regzip the contents into a
>         # temporary directory, and upload that with a special content encoding
> ```

Reply:
Thank you, I did not know that.